src/ostree/ostree-trivial-httpd.c

```diff
@@ -590,7 +590,7 @@ run (int argc, char **argv, GCancellable *cancellable, GError **error)
 
   if (!(opt_random_500s_percentage >= 0 && opt_random_500s_percentage <= 99))
     {
-      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_BUSY,
```

Comment:
You can just drop this part. As mentioned in the PR conversation, it's just coincidental that it also uses `G_IO_ERROR_FAILED`. This represents an error in the HTTP server if you give it a bogus `--random-500s` and doesn't reflect what error the HTTP client will receive. The only changes needed are on the client side.